File: apps/workbench/app/models/api_client_authorization.rb
# Copyright (C) The Arvados Authors. All rights reserved.
#
# SPDX-License-Identifier: AGPL-3.0

class ApiClientAuthorization < ArvadosBase
  def editable_attributes
    %w(expires_at default_owner_uuid)
  end
  def self.creatable?
    false
  end
end

File: src/models/api-client-authorization.ts
// Copyright (C) The Arvados Authors. All rights reserved.
//
// SPDX-License-Identifier: AGPL-3.0

import { Resource } from '~/models/resource';

export interface ApiClientAuthorization extends Resource {
    uuid: string;
    apiToken: string;
    apiClientId: number;
    userId: number;
    createdByIpAddress: string;
    lastUsedByIpAddress: string;
    lastUsedAt: string;
    expiresAt: string;
    createdAt: string;
    updatedAt: string;
    ownerUuid: string;
    defaultOwnerUuid: string;
    scopes: string[];
}

export const getTokenV2 = (aca: ApiClientAuthorization): string =>
    `v2/${aca.uuid}/${aca.apiToken}`;